controllers/oauthAuthentication.js

```diff
@@ -34,64 +38,53 @@ export const oauthLoginAuthorize = (req, res) => {
 
 export const oauthLoginCallback = async (req, res, next) => {
   try {
-    // Set request params
-    const body = {
-      grant_type: 'authorization_code',
-      redirect_uri: `${config.FS_URL}${config.LOGIN_CALLBACK_FS_PATH}`,
-      client_id: config.AUTHENTICATION_CLIENT_ID,
-      client_secret: config.AUTHENTICATION_CLIENT_SECRET,
-      code: req.query.code,
+    const spConfig = {
+      clientId: config.AUTHENTICATION_CLIENT_ID,
+      clientSecret: config.AUTHENTICATION_CLIENT_SECRET,
+      redirectUri: `${config.FS_URL}${config.LOGIN_CALLBACK_FS_PATH}`,
     };
 
-    // Request access token.
-    const { data: { access_token: accessToken, id_token: idToken } } = await httpClient({
-      method: 'POST',
-      headers: { 'content-type': 'application/x-www-form-urlencoded' },
-      data: querystring.stringify(body),
-      url: `${config.FC_URL}${config.TOKEN_FC_PATH}`,
-    });
-
-    if (!accessToken || !idToken) {
-      return res.sendStatus(401);
+    const { statusCode, user, idToken = null } = await requestUserInfo(req, spConfig);
```

Comment:
```suggestion
    const { statusCode, user, idToken = null } = await requestUserInfo( spConfig);
```
Vire le req et passe le code directement dans les spConfig.

Reply:
J ai mis le requestToken à part